filecache.go

```diff
@@ -15,6 +17,10 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+type Downloader interface {
```

Comment:
You don't need this interface, there is already a DownloadFunc that's public on this object. If you want to pass the downloader or a context in here you can do it in the closure.

Reply:
Oh, indeed. Not sure why I didn't do that :)